### PR TITLE
save tests result in separate json file

### PR DESCRIPTION
**Pull request by AlexaZhuk**

### CODE REVIEW

File: QuizApp/Results/TestResult.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace QuizApp.Results
+{
+	internal class TestResult
+	{
+		public int RightAnswers { get; set; }
+		public int WrongAnswers { get; set; }
+
```
Comment (MrSantiaga) [inline QuizApp/Results/TestResult.cs]: extra line or separate props by lines

File: QuizApp/Program.cs [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
 		static void Main(string[] args)
 		{
+			Console.WriteLine("Hallo, what is your name?");
+			string userName = Console.ReadLine();
 			string json = File.ReadAllText("C:\\Users\\Alexa\\Desktop\\test\\q.json");
 			AllQuestions questionsData = JsonConvert.DeserializeObject<AllQuestions>(json);
 
 			int rightAnswersCount = 0;
+			List<AnswerDetail> userAnswers = new List<AnswerDetail>();
+
```
Comment (MrSantiaga) [inline QuizApp/Program.cs]: extra line

File: QuizApp/Program.cs [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
 	{
 		static void Main(string[] args)
 		{
+			Console.WriteLine("Hallo, what is your name?");
+			string userName = Console.ReadLine();
 			string json = File.ReadAllText("C:\\Users\\Alexa\\Desktop\\test\\q.json");
 			AllQuestions questionsData = JsonConvert.DeserializeObject<AllQuestions>(json);
 
 			int rightAnswersCount = 0;
+			List<AnswerDetail> userAnswers = new List<AnswerDetail>();
```
Comment (MrSantiaga) [inline QuizApp/Program.cs]: userAnswers -> details

File: QuizApp/Results/TestResult.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace QuizApp.Results
+{
+	internal class TestResult
```
Comment (MrSantiaga) [inline QuizApp/Results/TestResult.cs]: TestResult -> QuizResult

File: QuizApp/Program.cs [hunk 5]
```diff
@@ -25,13 +31,35 @@ static void Main(string[] args)
...
+			testResult.Details = userAnswers;
+
+			string path = $"C:\\Users\\Alexa\\Desktop\\test\\{userName}_{DateTime.Now:dd-MM-yyyy_hh-mm-ss}.json";
+			string resultTry = JsonConvert.SerializeObject(testResult);
+			File.WriteAllText(path, resultTry);
+
+
+
 		}
+
```
Comment (MrSantiaga) [inline QuizApp/Program.cs]: extra line

File: QuizApp/Program.cs [hunk 6]
```diff
@@ -25,13 +31,35 @@ static void Main(string[] args)
...
+
```
Comment (MrSantiaga) [inline QuizApp/Program.cs]: extra line

File: QuizApp/Program.cs [hunk 7]
```diff
@@ -25,13 +31,35 @@ static void Main(string[] args)
...
+			TestResult testResult = new TestResult();
+			testResult.WrongAnswers = questionsData.Questions.Count - rightAnswersCount;
+			testResult.RightAnswers = rightAnswersCount;
+			testResult.Details = userAnswers;
+
+			string path = $"C:\\Users\\Alexa\\Desktop\\test\\{userName}_{DateTime.Now:dd-MM-yyyy_hh-mm-ss}.json";
+			string resultTry = JsonConvert.SerializeObject(testResult);
+			File.WriteAllText(path, resultTry);
+
```
Comment (MrSantiaga) [inline QuizApp/Program.cs]: extra line

File: QuizApp/Program.cs [hunk 8]
```diff
@@ -25,13 +31,35 @@ static void Main(string[] args)
...
+
```
Comment (MrSantiaga) [inline QuizApp/Program.cs]: extra line

File: QuizApp/Program.cs [hunk 9]
```diff
@@ -25,13 +31,35 @@ static void Main(string[] args)
...
+
```
Comment (MrSantiaga) [inline QuizApp/Program.cs]: extra line

File: QuizApp/Results/AnswerDetail.cs [hunk 10]
```diff
@@ -0,0 +1,8 @@
+﻿namespace QuizApp.Results
+{
+	internal class AnswerDetail
+	{
+		public string Question { get; set; }
+		public bool IsRight { get; set; }
```
Comment (MrSantiaga) [inline QuizApp/Results/AnswerDetail.cs]: IsRight -> IsCorrect

File: QuizApp/Results/TestResult.cs [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+	{
+		public int RightAnswers { get; set; }
```
Comment (MrSantiaga) [inline QuizApp/Results/TestResult.cs]: RightAnswers -> CorrectAnswers

**Review by MrSantiaga — CHANGES_REQUESTED** [review]

File: QuizApp/Program.cs (hunk 2 above)
Comment (AlexaZhuk) [reply QuizApp/Program.cs]: disagree

File: QuizApp/Results/TestResult.cs (hunk 11 above)
Comment (AlexaZhuk) [reply QuizApp/Results/TestResult.cs]: My little zanuda)

File: QuizApp/Program.cs (hunk 2 above)
Comment (MrSantiaga) [reply QuizApp/Program.cs]: > disagree

noone cares

**Review by MrSantiaga — APPROVED** [review]

**Pull request merged**